### PR TITLE
validating configuration values

### PR DESCRIPTION
**Pull request by keithbostic**
@agorrod, @sueloverso @michaelcahill: I was thinking a bit more about #1536 and #1739 today, and thought something like this change might improve things.

In short, you can specify a function in `dist/api_dist.py` that is subsequently called to validate configuration values. The only values I implemented were `huffman_key` and `huffman_value`, as a proof-of-concept, but it could certainly apply to other configuration values like `block_compressor`, or `collator`.

As I said, I think this makes things better, but it doesn't solve all of the hard cases. For example, I can't think of any reasonable way to catch:
```
self.session.create('table:xxx', 'type=lsm,lsm=(bloom_config=(columns=(FOO,BAR)))')
```
without actually creating the underlying files when the LSM tree is created.

If `session.create` is the real problem here: we could probably figure out a way to execute the full `session.create` path without actually creating the file. It wouldn't be pretty, but as I said, if `session.create` is the real problem, that might be another way to attack it.

In #1536, agorrod commented:
> Could we consider splitting out configuration validation from the various open methods?

and after poking around this morning, I'm much less optimistic that would be possible -- catching the above error without actually creating the file (or at least, pretending to create the file), seems almost impossible to me.

### CODE REVIEW

**Comment by keithbostic** [comment]
@michaelcahill, this branch has fixes for a problem discussed #1536 -- we weren't handling hexadecimal symbol/frequency values in Huffman files -- so I put them in this branch. If we don't take this branch, not a problem, I'll just cherry-pick them into develop.

File: src/btree/bt_huffman.c [hunk 1]
```diff
@@ -128,28 +128,67 @@ static const struct __wt_huffman_table __wt_huffman_nytenglish[] = {
...
+
+	/* Look for a prefix and file name. */
+	len = 0;
+	if (WT_PREFIX_MATCH(v->str, "utf8")) {
+		if (is_utf8p != NULL)
+			*is_utf8p = 1;
+		len = strlen("utf8");
+	} else if (WT_PREFIX_MATCH(v->str, "utf16")) {
+		if (is_utf8p != NULL)
+			*is_utf8p = 0;
```
Comment (sueloverso) [inline src/btree/bt_huffman.c]: These two lines should move outside the conditional up with `len=0;` so that we always initialize `is_utf8p` to something.

Comment (keithbostic) [inline src/btree/bt_huffman.c]: If `is_utf8p` isn't set, we're returning an error. Regardless, happy to change this if you like it better, I've pushed fix.

**Comment by michaelcahill** [comment]
Thanks, @keithbostic -- this lgtm, and we can add more checks for other config values after merging.  I'll merge.

**Pull request merged**